.github/workflows/ci-tests.yml

```diff
@@ -358,14 +358,81 @@ jobs:
           name: coverage-data
 
       - name: Combine coverage & fail if it's <100%.
+        id: report_coverage
+        # NOTE: Setting the pipefail option here means that even when
+        # piping the output to `tee`, we still get the exit code of the
+        # `coverage report` command.
         run: |
+          set -o pipefail
           python -m coverage combine
           python -m coverage html --skip-covered --skip-empty
-          python -m coverage report --fail-under=100
+          python -m coverage report --fail-under=100 --skip-covered --skip-empty -m | tee coverage-report.txt
 
       - name: Upload HTML report if check failed.
         uses: actions/upload-artifact@v3
         with:
           name: html-report
           path: htmlcov
         if: ${{ failure() }}
+
+      - name: Update PR Comment.
+        uses: actions/github-script@v6
+        # Run even if the coverage step failed.
+        if: always()
+        with:
+          script: |
+            // First list the existing comments
+
+            const trigger_str = 'Coverage Results'
+            console.log("Getting existing comments...")
+
+            const comments = await github.paginate(
+              github.rest.issues.listComments,
+              {
+                owner: 'sqlfluff',
+                repo: 'sqlfluff',
+                issue_number: context.issue.number
+              }
+            )
+
+            comment_id = null
+            console.log("Got %d comments", comments.length)
+
+            for (const comment of comments) {
+              if (comment.body.indexOf(trigger_str) >= 0) {
+                console.log("Found target comment ID: %d", comment.id)
+                comment_id = comment.id
+              } else {
+                console.log("Comment ID %d not valid with body:\n%s.", comment.id, comment.body)
+              }
+            }
+
+            previous_outcome = "${{ steps.report_coverage.outcome }}"
+            console.log("Previous coverage step outcome: %s", previous_outcome)
+            if (previous_outcome == "success") {
+              status_emoji = "✅"
+            } else {
+              status_emoji = "⚠️"
+            }
+
+            const { promises: fs } = require('fs')
+            let content = await fs.readFile('coverage-report.txt', 'utf8')
```

Comment:
`content` could be a `const`

Reply:
On this one, I'm not sure I really get the impact of `let`, `const` and perhaps their combination. When should I use each?

Reply:
`foo = true` could work for either setting or changing a variable, but `let foo = true` is saying that this is the first time we are setting this (and might help you find accidental variable shadowing)

`const foo = true` means `foo` cannot change later